### PR TITLE
Update bottlerocket-sdk to v0.46.0

### PR DESCRIPTION
**Pull request by ginglis13**
**Issue number:**

n/a

**Description of changes:**

**Description of changes:**

- Updates Bottlerocket SDK version to 0.46.0.

**Testing done:**

- [x] build kit on aarch64 host
- [x] build kit on x86_64 host
- [x] build Bottlerocket with new kit on x86_64 host for aarch64
- [x] build Bottlerocket with new kit on x86_64 host for x86_64
- [x] build Bottlerocket with new kit on aarch64 host for aarch64
- [x] build Bottlerocket with new kit on aarch64 host for x86_64
- [x] launched a x86 aws-k8s-1.30 instance.

```
[ssm-user@control]$ apiclient get os
{
  "os": {
    "arch": "x86_64",
    "build_id": "e141c066-dirty",
    "pretty_name": "Bottlerocket OS 1.25.0 (aws-k8s-1.30)",
    "variant_id": "aws-k8s-1.30",
    "version_id": "1.25.0"
  }
}
```

**Terms of contribution:**

By submitting this pull request, I agree that this contribution is dual-licensed under the terms of both the Apache License, version 2.0, and the MIT license.

### CODE REVIEW

**Comment by ginglis13** [comment]
^ add commit to use new cargo macros from the SDK in `os` and `early-boot-config`

**Comment by ginglis13** [comment]
^ rebase to remove temp commit and reference public bottlerocket-sdk:v0.46.0

**Comment by ginglis13** [comment]
CI failing due to clippy lints from the SDK's update to rust 1.81.0. We're using the SDK rust for the `check-clippy` task: https://github.com/bottlerocket-os/twoliter/blob/929a51756ab06b57e22f67c6bf4f08538ab41487/twoliter/embedded/Makefile.toml#L464

I'm going to disable the CI for this PR to unblock and create an issue to track. We should probably have some static clippy configuration in the core kit to avoid this in the future.

**Comment by ginglis13** [comment]
^ re-enable check-clippy in CI, fix lints via `cargo clippy --fix`

Manual changes to fix lints in documentation indents, and other manual changes in:

```
sources/api/apiserver/src/server/exec/child.rs
sources/api/apiserver/src/server/mod.rs
sources/netdog/src/interface_id.rs
```

**Comment by ginglis13** [comment]
^ some of the autofixes from clippy weren't `rustfmt`'d 🤦

File: sources/api/schnauzer/src/helpers/mod.rs [hunk 1]
```diff
@@ -2299,10 +2299,10 @@ mod test_replace_ipv4_octet {
 
     #[test]
     fn test_invalid_replace_ipv4_octet() {
-        let test_cases = vec![
+        let test_cases = [
             json!({"ip": "192.168.1.1", "index": 4, "value": "10"}), // Invalid index
             json!({"ip": "invalid-ip", "index": 3, "value": "10"}),  // Invalid IP
-            json!({"ip": "192.168.1.1", "index": 3, "value": "257"}), // Invalid octet value
+            json!({"ip": "192.168.1.1", "index": 3, "value": "257"}),
```
Comment (cbgbt) [inline sources/api/schnauzer/src/helpers/mod.rs]: nit: we may wish to keep this comment

**Review by cbgbt — APPROVED** [review]

**Review by yeazelm — APPROVED** [review]

**Review by arnaldo2792 — APPROVED** [review]

**Pull request merged**